common/src/main/java/io/homeassistant/companion/android/common/data/prefs/WearPrefsRepositoryImpl.kt

```diff
@@ -51,6 +53,26 @@ class WearPrefsRepositoryImpl @Inject constructor(
 
                 localStorage.putInt(MIGRATION_PREF, MIGRATION_VERSION)
             }
+
+            if (currentVersion == 1) {
+                val template = localStorage.getString(legacyPrefTileTemplate)
+                val templateRefreshInterval = localStorage.getInt(
+                    legacyPrefTileTemplateRefreshInterval
+                )
+
+                if (template != null && templateRefreshInterval != null) {
+                    val templates = mapOf(
+                        null to TemplateTileData(template, templateRefreshInterval).toJSONObject()
+                    )
+
+                    localStorage.putString(PREF_TILE_TEMPLATES, JSONObject(templates).toString())
+                }
+
+                localStorage.remove(legacyPrefTileTemplate)
+                localStorage.remove(legacyPrefTileTemplateRefreshInterval)
+
+                localStorage.putInt(MIGRATION_PREF, MIGRATION_VERSION)
+            }
```

Comment:
Not fully convinced whether this is the right way forward yet. Having a single entry for all the templates and their respective refresh intervals feels somewhat logical though.

Reply:
Why are you adding a migration here _and_ doing the runCatching/fold thing in the individual preference? Having it in one place should be enough (and I think a migration here is nicest, so you can avoid complicated code in the individual preference functions).

Reply:
Good question! The migration should indeed be enough. I started out with a similar approach as #3697 (hence  `runCatching` etc), but I cleaned it up now. Please take a look.